File: Search Insert Position - 35/ans.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int start = 0, end = nums.size() - 1;
        int mid;
        while (start <= end){
            mid = (end + start) / 2;
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                end = mid - 1;
            else
                start = mid + 1;
        }
        return start;
    }
};

File: Remove Element - 27/ans.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int c = 0;
        for (int i = 0; i < nums.length; ++i){
            if (nums[i] != val){
                int j = i;
                c++;
                int value = nums[j];
                while (j > 0 && nums[j - 1] == val){
                    nums[j] = nums[j - 1];
                    j--;
                }
                nums[j] = value;
            }
        }
        return c;
    }
}
